app/uk/gov/hmrc/merchandiseinbaggage/model/core/ThresholdAllowance.scala

```diff
@@ -27,14 +28,32 @@ object ThresholdAllowance {
 
   private[core] val formatter = "%,.2f"
 
+  //TODO this should be all done in BE
   implicit class ThresholdAllowanceLeft(allowance: ThresholdAllowance) {
     import allowance._
     def allowanceLeft: Double =
-      Try {
-        val sum = calculationResponse.results.calculationResults.map(_.gbpAmount.value).sum
-        (destination.threshold.value.toDouble - sum.toDouble) / 100
-      }.getOrElse((destination.threshold.value - calculationResponse.results.calculationResults.map(_.gbpAmount.value).sum) / 100)
+      goods.goods.headOption
+        .map { g =>
+          calculateAllowanceLeft(goods, calculationResponse, destination, g)
+        }
+        .getOrElse(0)
 
     def toUIString: String = s"£${formatter.format(allowanceLeft)}"
   }
+
+  private def calculateAllowanceLeft(
+    goods: DeclarationGoods,
+    calculationResponse: CalculationResponse,
+    destination: GoodsDestination,
+    good: Goods): Double =
+    Try {
+      good match {
+        case _: ImportGoods =>
+          val sum = calculationResponse.results.calculationResults.map(_.gbpAmount.value).sum
```

Comment:
Does this include amendments?  if not should it?

Reply:
not knowledge of new/amend here. This calculate based on goods regardless if are amend or new

Reply:
i think Andy is asking about are we considering amendments too for calculating the total purchaseDetails ?